cpp/src/spatial/point_in_polygon.cu

```diff
@@ -0,0 +1,233 @@
+        auto type = cudf::data_type{ tid };
+        auto results = cudf::make_fixed_width_column(type,
+                                                     size,
+                                                     cudf::mask_state::UNALLOCATED,
+                                                     stream,
+                                                     mr);
+
+        if (results->size() == 0)
+        {
+            return results;
+        }
+
+        constexpr cudf::size_type block_size = 256;
+
+        cudf::experimental::detail::grid_1d grid{results->size(), block_size, 1};
+
+        auto kernel = point_in_polygon_kernel<T>;
+
+        kernel<<<grid.num_blocks, block_size, 0, stream>>>(
+            test_points_x.size(),
+            test_points_x.begin<T>(),
+            test_points_y.begin<T>(),
+            poly_offsets.size(),
+            poly_offsets.begin<cudf::size_type>(),
+            poly_ring_offsets.begin<cudf::size_type>(),
+            poly_points_x.begin<T>(),
+            poly_points_y.begin<T>(),
+            results->mutable_view().begin<cudf::size_type>()
+        );
+
+        return results;
+    }
+};
+
+} // anonymous namespace
+
+namespace cuspatial {
+
+namespace detail {
+
+std::unique_ptr<cudf::column>
+point_in_polygon(cudf::column_view const& test_points_x,
+                 cudf::column_view const& test_points_y,
+                 cudf::column_view const& poly_offsets,
+                 cudf::column_view const& poly_ring_offsets,
+                 cudf::column_view const& poly_points_y,
+                 cudf::column_view const& poly_points_x,
+                 rmm::mr::device_memory_resource* mr,
+                 cudaStream_t stream)
+{
+    return cudf::experimental::type_dispatcher(test_points_x.type(), point_in_polygon_functor(),
+                                               test_points_x,
+                                               test_points_y,
+                                               poly_offsets,
+                                               poly_ring_offsets,
+                                               poly_points_y,
+                                               poly_points_x,
+                                               mr,
+                                               stream);
+}
+
+}
+
+std::unique_ptr<cudf::column>
+point_in_polygon(cudf::column_view const& test_points_x,
+                 cudf::column_view const& test_points_y,
+                 cudf::column_view const& poly_offsets,
+                 cudf::column_view const& poly_ring_offsets,
+                 cudf::column_view const& poly_points_y,
+                 cudf::column_view const& poly_points_x,
+                 rmm::mr::device_memory_resource* mr)
+{
+    CUSPATIAL_EXPECTS(test_points_x.size() == test_points_x.size() and
+                      poly_points_x.size() == poly_points_y.size(),
+                      "All points must have both x and y values");
```

Comment:
Typo (x == x) -- must not be testing this expectation. Also I don't think separate expectations should be combined, makes testing harder.

```suggestion
    CUSPATIAL_EXPECTS(test_points_x.size() == test_points_y.size(),
                      "Size mismatch between x and y test points");
    CUSPATIAL_EXPECTS(poly_points_x.size() == poly_points_y.size(),
                      "Size mismatch between x and y polygon points");
```